web-gateway/src/main/java/controllers/AbstractController.java

```diff
@@ -47,17 +50,21 @@ public AbstractController(MessagesApi messagesApi, UserService userService) {
     }
 
     protected CompletionStage<Nav> loadNav(Optional<UUID> userId) {
-        return userService.getUsers(Optional.of(0), Optional.of(10)).invoke().thenApply(users -> {
-            Optional<User> currentUser = userId.flatMap(id -> {
-                for (User u: users.getItems()) {
-                    if (u.getId().equals(id)) {
-                        return Optional.of(u);
-                    }
+        CompletionStage<User> u = userService.getUser(userId.get())
+                .handleRequestHeader(authenticate(userId.get())).invoke();
+        CompletionStage<PaginatedSequence<User>> users = userService.getUsers(Optional.of(0), Optional.of(10))
+                .handleRequestHeader(authenticate(userId.get())).invoke();
+        return users.thenCombineAsync(u, (userPaginatedSequence, user) -> {
+            Optional<User> currentUser = userId.flatMap(uuid -> {
+
+                if (user.getId().equals(uuid)) {
+                    return Optional.of(user);
                 }
+
                 return Optional.empty();
             });
```

Comment:
With the change described above you can be sure that `userId` is present and use `get` here instead of `flatMap`.
Also, I don't understand in what case would `user.getId().equals(uuid)` return `false`.